query-engine/js-connectors/src/proxy.rs

```diff
@@ -301,7 +303,17 @@ impl From<JSResultSet> for QuaintResultSet {
             })
             .collect::<Vec<_>>();
 
-        QuaintResultSet::new(column_names, quaint_rows)
+        let last_insert_id = last_insert_id.and_then(|id| id.parse::<u64>().ok());
+        let mut quaint_result_set = QuaintResultSet::new(column_names, quaint_rows);
+
+        // Not a fan of this (extracting the `Some` value from an `Option` and pass it to a method that creates a new `Some` value),
+        // but that's Quaint's ResultSet API and that's how the MySQL connector does it.
+        // Sqlite, on the other hand, uses a `last_insert_id.unwrap_or(0)` approach.
```

Comment:
Just a guees, but I thing for MySQL, depending on your settings, 0 can actually be a legit value for last insert it.

Reply:
I believe you're referring to SQLite rather than MySQL, but I totally agree, and that's part of the reason why I've highlighted it in the comment. Perhaps @Weakky knows more about the reasons behind it

Reply:
I don't really have an opinion here, it's an implementation detail tbh. I guess asking for a u64 instead of an Option<u64> forces you to be explicit about what you want to pass. In the case of SQLite, we're explicitly always passing an id, even when there's none. In the case of MySQL, we're passing it only when we have one.